From f4c2bfd8fbed1fcf8d33abf54f614594779f57e8 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Thu, 11 Mar 2010 14:21:04 +0800
Subject: [PATCH] ENGR00121504 Revert ENGR00120191 to restore M4IF values

CR 00120191 is used to adjust IPU/VPU priority for memory
access in order to improve video performance. However this
change caused video playback and record unstable (See
CR 00120697, CR 00121478). It's decided to remove CR 00121504
change for the stability.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 board/freescale/mx51_3stack/lowlevel_init.S |    8 ++++----
 board/freescale/mx51_bbg/lowlevel_init.S    |    8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/board/freescale/mx51_3stack/lowlevel_init.S b/board/freescale/mx51_3stack/lowlevel_init.S
index b867335..b473f06 100644
--- a/board/freescale/mx51_3stack/lowlevel_init.S
+++ b/board/freescale/mx51_3stack/lowlevel_init.S
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2007, Guennadi Liakhovetski <lg@denx.de>
  *
- * (C) Copyright 2009 Freescale Semiconductor, Inc.
+ * (C) Copyright 2009-2010 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License as
@@ -89,16 +89,16 @@
 	 */
 	ldr r0, =M4IF_BASE_ADDR
 
-	ldr r1, =0x00000103
+	ldr r1, =0x00000203
 	str r1, [r0, #0x40]
 
 	ldr r1, =0x0
 	str r1, [r0, #0x44]
 
-	ldr r1, =0x00240126
+	ldr r1, =0x00120125
 	str r1, [r0, #0x9C]
 
-	ldr r1, =0x00230185
+	ldr r1, =0x001901A3
 	str r1, [r0, #0x48]
 .endm /* init_m4if */
 
diff --git a/board/freescale/mx51_bbg/lowlevel_init.S b/board/freescale/mx51_bbg/lowlevel_init.S
index 62e46dc..f7f780a 100644
--- a/board/freescale/mx51_bbg/lowlevel_init.S
+++ b/board/freescale/mx51_bbg/lowlevel_init.S
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2007, Guennadi Liakhovetski <lg@denx.de>
  *
- * (C) Copyright 2009 Freescale Semiconductor, Inc.
+ * (C) Copyright 2009-2010 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License as
@@ -88,16 +88,16 @@
 	 */
 	ldr r0, =M4IF_BASE_ADDR
 
-	ldr r1, =0x00000103
+	ldr r1, =0x00000203
 	str r1, [r0, #0x40]
 
 	ldr r1, =0x0
 	str r1, [r0, #0x44]
 
-	ldr r1, =0x00240126
+	ldr r1, =0x00120125
 	str r1, [r0, #0x9C]
 
-	ldr r1, =0x00230185
+	ldr r1, =0x001901A3
 	str r1, [r0, #0x48]
 
 /*
-- 
1.5.4.4

